app/controllers/items/new.js

```diff
@@ -595,6 +630,17 @@ export default GoodcityController.extend(
         });
       },
 
+      setPrinterValue(value) {
+        let printerId = value.id;
+        this.set("selectedPrinterDisplay", {
+          name: this.get("store")
+            .peekRecord("printer", printerId)
+            .get("name"),
+          id: printerId
+        });
```

Comment:
This `set` is the one I'm referring to in the comment above. It doesn't make sense since you're already building that object ( {name, id} ) in the computation

```suggestion
this.set('selectedPrinterId', printerId);
```